src/OrchardCore.Modules/OrchardCore.Lucene/Controllers/SearchController.cs

```diff
@@ -155,7 +158,8 @@ public async Task<IActionResult> Search(SearchIndexViewModel viewModel, PagerSli
                     .Take(pager.PageSize + 1);
             }
 
-            var containedItems = await queryDb.ListAsync();
+            // Sort the content items by their rank in the search results returned by Lucene.
+            var containedItems = (await queryDb.ListAsync()).OrderBy(x => contentItemIds.IndexOf(x.ContentItemId));
```

Comment:
I'd be interested to know if this could be part of the sql query above, i.e. sorted at source, rather than being done by linq afterwards.

Reply:
> I'd be interested to know if this could be part of the sql query above, i.e. sorted at source, rather than being done by linq afterwards.

I don't think it is possible to sort the search results in the SQL query using YesSql. Since the order of the search results is not predictable, a custom sort would need to be used. My understanding is that this is usually accomplished with a `CASE` statement:
```sql
ORDER BY CASE WHEN ContentItemId = '<ID of 1st result>' THEN '1'
              WHEN ContentItemId = '<ID of 2nd result>' THEN '2'
              WHEN ContentItemId = '<ID of 3rd result>' THEN '3'
              ...
              END ASC
```

I don't know of a way to generate custom orderings like that in YesSql.

I also noticed that [`GetAsync(IEnumerable<string> contentItemIds, bool latest = false)`](https://github.com/OrchardCMS/OrchardCore/blob/a747017be08d5839223ee4debfa028c7ac426f6e/src/OrchardCore/OrchardCore.ContentManagement/DefaultContentManager.cs#L94) orders the results of its SQL query using Linq.

Reply:
I think that's okay to not do ordering on YesSql side as it would not reduce the number of returned items.

Reply:
Yeah, I agree, I was just curious if you could. Then I googled it and looked at the sql required...